timesketch/frontend/src/components/Explore/TimelineChip.vue

```diff
@@ -14,28 +14,320 @@ See the License for the specific language governing permissions and
+        v-if="datasourceErrors.length"
+        class="b-tooltips"
+        style="padding-right: 6px; padding-left: 12px;"
+        @click.stop
+      >
+        <b-tooltip :label="failedImportsMessage" :type="isDarkTheme ? 'is-dark' : 'is-light'">
+          <span class="icon is-small" style="color:orange;">
+            <i class="fas fa-exclamation-triangle" @click="showInfoModal = !showInfoModal"></i>
+          </span>
+        </b-tooltip>
+      </span>
+      <span
+        class="tag is-small"
+        style="margin-left:10px;margin-right:-7px;background-color: rgba(255,255,255,0.5);min-width:50px;"
+      >
+        <span v-if="isSelected && !isEmptyState">{{
+          eventsCount | compactNumber
+        }}
+        </span>
+      </span>
+
+      <span v-if="meta.permissions.write" @click.stop>
+        <!-- The color change element -->
+        <b-dropdown class="color-picker-dropdown" append-to-body >
+          <template #trigger>
+            <a role="button" ref="colorPicker"></a>
+          </template>
+          <b-dropdown-item aria-role="menu" paddingless>
+            <color-picker v-model="initialColor" @input="updateColor"></color-picker>
+          </b-dropdown-item>
+        </b-dropdown>
+
+        <!-- 3-dots dropdown menu -->
+        <b-dropdown append-to-body>
+          <template #trigger>
+            <a role="button">
+              <i class="fas fa-ellipsis-v" style="padding-left: 14px;padding-right: 6px;"></i>
+            </a>
+          </template>
+          <b-dropdown-item aria-role="listitem" @click="showInfoModal = !showInfoModal">
+            <span class="icon is-small"><i class="fas fa-info-circle"></i></span>
+            <span>Info</span>
+          </b-dropdown-item>
+          <b-dropdown-item aria-role="listitem" v-if="timelineStatus === 'ready'" @click="showEditModal = !showEditModal">
+            <span class="icon is-small">
+            <i class="fas fa-edit"></i>
+            </span>
+            <span>Rename</span>
+          </b-dropdown-item>
+          <b-dropdown-item aria-role="listitem" v-if="timelineStatus === 'ready'" @click="showColorPicker()">
+            <span class="icon is-small">
+            <i class="fas fa-palette"></i>
+            </span>
+            <span>Change color</span>
+          </b-dropdown-item>
+          <b-dropdown-item aria-role="listitem" v-if="timelineStatus === 'ready'" @click="showAnalyzerModal = !showAnalyzerModal">
+            <span class="icon is-small">
+            <i class="fas fa-history"></i>
+            </span>
+            <span>Analyzer logs</span>
+          </b-dropdown-item>
+          <b-dropdown-item aria-role="listitem" @click="remove()" class="is-danger">
+            <span class="icon is-small">
+            <i class="fas fa-trash"></i>
+            </span>
+            <span>Delete</span>
+          </b-dropdown-item>
+        </b-dropdown>
+      </span>
+    </span>
   </div>
 </template>
 
 <script>
+import Vue from 'vue'
```

Comment:
Shouldn't the Vue package already be imported?

Reply:
Thanks Alex for pointing this out! I copied this part of code from one of the Timelines components so I didn't question it too much. It turns out you were right, removing that import doesn't cause any issues.

Reply:
I also searched for `Vue.set()` and `Vue.use()` across the code and it seems that this is the only time that we use `Vue.set()` outside of the `store.js` file, which is probably why we have to import Vue. There might be a better way of calling `Vue.set()` that doesn't require this file to import Vue, but that would probably need to wait for Johan to be back :)